pyEpiabm/pyEpiabm/sweep/host_progression_sweep.py

```diff
@@ -15,6 +17,18 @@ class HostProgressionSweep(AbstractSweep):
     and time to next infection status change.
     """
 
+    def __init__(self):
+        """Initialise parameters to be used in class methods. State
+        transition matrix is set where each row of the matrix corresponds
+        to a current infection status of a person. The columns of that
+        row then indicate the transition probabilities for the remaining
+        infection statuses. Number of infection states is also set by
+        taking the size of the InfectionStatus enum.
+        """
+        self.state_transition_matrix = \
+            pe.Parameters.instance().state_transition_matrix
+        self.number_of_states = len(InfectionStatus)
```

Comment:
I think it might be sensible to check these dimensions match on initialisation (that transition matrix is N x N for `N = len(InfectionStatus)`